# Generated by Django 2.2.10 on 2020-06-05 17:37

from django.apps import apps
from django.db import migrations
from django.db.utils import IntegrityError
from django.conf import settings
from django.core.exceptions import ImproperlyConfigured

def update_site_forward(apps, schema_editor):
    Sitio = apps.get_model('siteprofile', 'Sitio')
    for host in settings.ALLOWED_HOSTS:
        name=host.split('.')[0]
        if f'radon.{name}.apps.{name[0].upper()}{name[1:]}Config' in settings.INSTALLED_APPS:
            Sitio.objects.get_or_create(
                domain=host,
                name=name,
                modulo='radon.'+name+'.urls'
            )
        else:
            raise ImproperlyConfigured(f'Â¡La app {name} no existe en el proyecto!')


class Migration(migrations.Migration):

    dependencies = [
        ('siteprofile', '0001_initial'),
    ]

    operations = [
        migrations.RunPython(update_site_forward),
    ]
